libr/core/rvc.c

```diff
@@ -733,30 +733,40 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
 		eprintf ("Can't commit");
 		return false;
 	}
-	if (!message) {
+	if (R_STR_ISEMPTY (message)) {
 		char *path = NULL;
 		(void)r_file_mkstemp ("rvc", &path);
 		if (path) {
 			char *editor = r_sys_getenv ("EDITOR");
 			if (!R_STR_ISEMPTY (editor)) {
-				if (!r_sys_cmdf ("%s %s", editor, path)) {
+				if (r_sys_cmdf ("%s %s", editor, path)) {
+					r_file_rm (path);
 					free (path);
 					free (editor);
 					return false;
-				} else {
-					free (editor);
-					free (path);
-					return false;
 				}
+			} else {
+				eprintf ("Please set the $EDITOR env var\n");
```

Comment:
Better inform the user about cfg.editor r2 var for portability

Reply:
Well actually r2 have a line editor like ed in rcons and this should be the spawned editor if none is set

Reply:
how can I spawn it?

Reply:
seems like it wont work if cbeditor is set. ill fix in few mins. hold on